src/main/java/apoc/custom/CypherProcedures.java

```diff
@@ -126,8 +133,16 @@ public boolean registerProcedure(@Name("name") String name, @Name("statement") S
                         Mode.valueOf(mode.toUpperCase()), admin, null, new String[0], description, null, false, true
                 );
                 procedures.register(new CallableProcedure.BasicProcedure(signature) {
+                    private CustomStatementRegistry registry = self;
                     @Override
                     public RawIterator<Object[], ProcedureException> apply(org.neo4j.kernel.api.proc.Context ctx, Object[] input, ResourceTracker resourceTracker) throws ProcedureException {
+                        GraphPropertiesProxy properties = getProperties(api);
+                        long proceduresLastUpdate = getProceduresLastUpdate(properties);
+                        if (proceduresLastUpdate > PROCEDURES_LAST_UPDATE.get()) {
```

Comment:
can we move the check into a separate function.
they should always be called not just in the if case.